### PR TITLE
Refactor openssl FIPS hash tests to be generic

### PR DESCRIPTION
**Pull request by ilmanzo**
- Related ticket: https://progress.opensuse.org/issues/159780
- Needles: no
- Verification runs:
	- https://openqa.suse.de/tests/14718572
	- https://openqa.suse.de/tests/14718565
	- https://openqa.suse.de/tests/14718553
	- https://openqa.suse.de/tests/14718554

### CODE REVIEW

**Comment by pdostal** [comment]
Hi Andrea, you `use openssl_fips_hash_common;` so it expects `lib/openssl_fips_hash_common.pm`.

I suggest to move `tests/fips/openssl/openssl_fips_hash_common.pm` to `openssl_fips_hash_common.pm`.

**Comment by ilmanzo** [comment]
> Hi Andrea, you `use openssl_fips_hash_common;` so it expects `lib/openssl_fips_hash_common.pm`.
> 
> I suggest to move `tests/fips/openssl/openssl_fips_hash_common.pm` to `lib/openssl_fips_hash_common.pm`.

Thanks @pdostal I will try; oddly enough, the test suite runs without problems, so my doubt is about the best practice to follow: `openssl_fips_hash_common` is used only by two modules, as it isn't a generic library. 

It could be an option to have it "near" (in the same path) of the modules using it, also because I don't want to overcrowd an already packed `lib` directory ...

**Comment by pdostal** [comment]
I don't know but I'd personally follow the structure. As this file is specific to security, I'd recommend to create `lib/security/` and place all your smaller and bigger libraries over there :smile:

**Review by pdostal — COMMENTED** [review]
Nice 🚀

File: lib/security/openssl_fips_hash_common.pm [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# Maintainer: QE Security <none@suse.de>
+
+package security::openssl_fips_hash_common;
+
+use base Exporter;
+use Exporter;
+
+use strict;
+use warnings;
+use testapi;
```
Comment (perlpunk) [inline lib/security/openssl_fips_hash_common.pm:16]: little bit too late, but:
```suggestion
use strict;
use warnings;
use testapi;

use base 'Exporter';
```

Comment (ilmanzo) [reply lib/security/openssl_fips_hash_common.pm:16]: Good point! Addressed on https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19597

**Pull request merged**